webapp/src/DataTransferObject/Shadowing/SubmissionEvent.php

```diff
@@ -14,6 +14,6 @@ public function __construct(
         public readonly string $teamId,
         public readonly string $time,
         public readonly ?string $entryPoint,
-        public readonly array $files,
+        public readonly ?array $files,
```

Comment:
Same as above: per spec this is required, so we shouldn't merge this in `main`.

Reply:
So when we are analyst and have the analysts access policy:
We start with access:
```
{
   "endpoints": [
     { "type": "submissions", "properties": ["id","files",...]}
   ]
}
```

But during the freeze we don't have that access anymore, what fields should the CDS give?

Reply:
Indeed, files are omitted during the freeze for some users, so it is not required. The spec is a bit weird in that it a field doesn’t have a ? it can still be omitted if the user doesn’t have access

Reply:
Hmmm, yes, that's a fair point. This is a part of the spec that's a bit annoying where things differ in a bit ugly way depending on your role and time in the contest.

So this can be left as is, but it would be good to add a note to the code as to why it's optional.

Reply:
Yeah adding a comment makes sense.